app/src/main/java/org/mozilla/fenix/customtabs/CustomTabToolbarIntegration.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.customtabs
+
+import android.content.Context
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.feature.toolbar.ToolbarFeature
+import mozilla.components.feature.toolbar.ToolbarPresenter
+import mozilla.components.lib.publicsuffixlist.PublicSuffixList
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.toolbar.MenuPresenter
+import org.mozilla.fenix.components.toolbar.ToolbarIntegration
+import org.mozilla.fenix.components.toolbar.ToolbarMenu
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.theme.ThemeManager
+
+class CustomTabToolbarIntegration(
+    context: Context,
+    toolbar: BrowserToolbar,
+    toolbarMenu: ToolbarMenu,
+    sessionId: String,
+    isPrivate: Boolean
+) : ToolbarIntegration(
+    context = context,
+    toolbar = toolbar,
+    sessionId = sessionId,
+    renderStyle = ToolbarFeature.RenderStyle.RegistrableDomain
+) {
+
+    private var renderStyle: ToolbarFeature.RenderStyle = ToolbarFeature.RenderStyle.RegistrableDomain
+
+    init {
+        toolbar.display.menuBuilder = toolbarMenu.menuBuilder
+        toolbar.private = isPrivate
+    }
+
+    private val toolbarPresenter: ToolbarPresenter = ToolbarPresenter(
+        toolbar,
+        context.components.core.store,
+        sessionId,
+        ToolbarFeature.UrlRenderConfiguration(
+            PublicSuffixList(context),
+            ThemeManager.resolveAttribute(R.attr.primaryText, context),
+            renderStyle = renderStyle
+        )
+    )
+
+    private var menuPresenter =
+        MenuPresenter(toolbar, context.components.core.sessionManager, sessionId)
+
+    override fun start() {
+        menuPresenter.start()
+        toolbarPresenter.start()
+    }
+
+    override fun stop() {
+        menuPresenter.stop()
+        toolbarPresenter.stop()
```

Comment:
And if we use the presenter in `ToolbarIntegration`, we don't need any of these, either.  That removes the entire implementation of this class, and `CustomTabToolbarIntegration` becomes:

```kotlin
class CustomTabToolbarIntegration(
    context: Context,
    toolbar: BrowserToolbar,
    toolbarMenu: ToolbarMenu,
    sessionId: String,
    isPrivate: Boolean
) : ToolbarIntegration(
    context = context,
    toolbar = toolbar,
    sessionId = sessionId,
    renderStyle = ToolbarFeature.RenderStyle.RegistrableDomain,
    isPrivate = isPrivate,
    toolbarMenu = toolbarMenu
)
```

I like this a lot, because it clarifies what really varies between the two classes (`CustomTabToolbarIntegration` uses a custom style, and `DefaultTabToolbarIntegration` includes extra setup steps).

Reply:
Thanks for catching this! Good to see lots of simplification as a result.